arangod/Pregel/Worker/Handler.h

```diff
@@ -49,18 +50,22 @@ struct VerticesProcessed {
 
 template<typename V, typename E, typename M, typename Runtime>
 struct WorkerHandler : actor::HandlerBase<Runtime, WorkerState<V, E, M>> {
+  DispatchStatus const& dispatchStatus =
+      this->template dispatch<pregel::message::StatusMessages>;
+  DispatchMetrics const& dispatchMetrics =
+      this->template dispatch<metrics::message::MetricsMessages>;
+  DispatchConductor const& dispatchConductor =
+      this->template dispatch<conductor::message::ConductorMessages>;
+
   auto operator()(message::WorkerStart start)
       -> std::unique_ptr<WorkerState<V, E, M>> {
     LOG_TOPIC("cd696", INFO, Logger::PREGEL) << fmt::format(
         "Worker Actor {} started with state {}", this->self, *this->state);
-    // TODO GORDO-1556
-    // _feature.metrics()->pregelWorkersNumber->fetch_add(1);
-    this->template dispatch<conductor::message::ConductorMessages>(
-        this->state->conductor, ResultT<conductor::message::WorkerCreated>{});
 
-    this->template dispatch<metrics::message::MetricsMessages>(
-        this->state->metricsActor,
-        arangodb::pregel::metrics::message::WorkerStarted{});
+    auto newState =
+        this->executionState->receive(this->sender, start, dispatchStatus,
+                                      dispatchMetrics, dispatchConductor);
+    changeState(newState);
```

Comment:
If you stay with the approach that you give all dispatch functions to the execution function receive method, you could actually change the execution state just inside the execution state because it knows the worker - then the receive function does not need to return anything.

Reply:
Agreed. Will change this, contingent on Markus letting us get away with 3 callbacks in the `receive` method.

Reply:
Bundled all dispatch callbacks (there were more) into a `Dispatcher` struct.